src/config.c

```diff
@@ -173,54 +163,151 @@ config_read_cmd(const char *ipath, const char *set_command_format)
     if (path == NULL) return NULL;
    }
 
-  set_command_size = snprintf(NULL,0,set_command_format,path)+1;
-  set_command = (char *)malloc(set_command_size);
-  if (set_command == NULL)
+  rha = config_new(path);
+  free(path);
+
+  if ( ! config_read_cmd(rha->config_path, set_command_format, rha) )
    {
-    free(path);
+    config_free(rha);
     return NULL;
    }
-  snprintf(set_command, set_command_size, set_command_format, path);
 
-  cf = popen(set_command, "r");
-  if (cf == NULL) 
+  config_dir = (char *)malloc(strlen(rha->config_path)+3);
+  sprintf(config_dir, "%s.d", rha->config_path);
+  file_cnt = scandir(config_dir, &file_list, NULL, alphasort);
+  for (i = 0; i < file_cnt; i++)
    {
-    free(set_command);
-    free(path);
-    return NULL;
+    if (file_list[i]->d_name[0] != '.')
+	 {
+      snprintf(full_file, PATH_MAX, "%s/%s", config_dir, file_list[i]->d_name);
```

Comment:
I guess it would be interesting to see what could happen if the user specifies a very long path for config_dir, ey?  :)

But i realize this is the blahp and we don't really worry about it.

Reply:
Yeah, I was being a little lazy here with the static buffer. Since it's non-privileged code, I'm not too worried about that.